api-2.0/src/test/java/org/openmrs/module/attachments/AttachmentsServiceTest.java

```diff
@@ -0,0 +1,75 @@
+package org.openmrs.module.attachments;
+
+import org.junit.Test;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.Visit;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.VisitService;
+import org.openmrs.module.attachments.obs.Attachment;
+import org.openmrs.module.attachments.obs.TestHelper;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class AttachmentsServiceTest extends BaseModuleContextSensitiveTest {
+	
+	@Autowired
+	private AttachmentsService as;
+	
+	@Autowired
+	protected TestHelper testHelper;
+	
+	@Autowired
+	@Qualifier("patientService")
+	protected PatientService ps;
+	
+	@Autowired
+	@Qualifier("visitService")
+	protected VisitService vs;
+	
+	@Autowired
+	@Qualifier("encounterService")
+	private EncounterService es;
+	
+	@Test
+	public void getAttachments_shouldReturnAttachments() throws Exception {
+		
+		// Setup
+		List<Obs> obsList = testHelper.saveComplexObs(3);
+		
+		Obs obs = obsList.get(0);
+		Patient patient = obs.getEncounter().getPatient();
+		Visit visit = obs.getEncounter().getVisit();
+		Encounter encounter = obs.getEncounter();
+		
+		// Replay
+		List<Attachment> attachmentsList = as.getAttachments(patient, visit, encounter, true);
+		
+		// Verify
+		List<Attachment> originalAttachmentsList = obsList.stream().map(Attachment::new).collect(Collectors.toList());
```

Comment:
Just extract arrays of UUIDs (so they will just be arrays of `String`, and `String` objects are natively comparable in Java) and compare them using `Assert.assertArrayEquals(..)`, example [here](https://github.com/openmrs/openmrs-module-webservices.rest/blob/ec0a7fad4522fc0c2bd346d5abec4eb9425a2ec3/omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ProviderController2_0Test.java#L66).

Reply:
Also by convention unit tests asserts compare _expected_ values vs _actual_ values. To stick to this convention I would refactor your lists that way:
- `attachmentsList` → `attachments` or `actualAttachments` 
- `originalAttachmentsList ` → `expectedAttachments` 

No need to suffix with _List_, a plural is enough for the reader to understand that it is a collection of objects.